mongo/index_view.go

```diff
@@ -31,9 +31,6 @@ var ErrInvalidIndexValue = errors.New("invalid index value")
 // ErrNonStringIndexName is returned if an index is created with a name that is not a string.
 var ErrNonStringIndexName = errors.New("index name must be a string")
```

Comment:
This error value appears to be unused. Can we mark it as deprecated, or should it be returned in some case?

Reply:
It seems not being used in the current v1 code base either. Is removing it considered a breaking change?

Reply:
Yeah, removing it is technically breaking because someone's code might reference the symbol, which would cause a compilation error if we remove it. The best we can do is mark it deprecated and note that it's no longer used.